orchestrator/cli/migrate_domain_models.py

```diff
@@ -89,7 +89,11 @@ def map_product_blocks_in_class(
         model_class._get_depends_on_product_block_types()
     )
 
-    new_blocks = {block.name: block for block in product_blocks_types_in_model if block.name not in product_blocks}
+    new_blocks = {
+        block.name: block
+        for block in product_blocks_types_in_model
+        if (block.name not in product_blocks) and (block.name is not None)
```

Comment:
Shouldn't we wonder why block.name can be None?

Reply:
This is a peculiarity in how we defined the ProductBlockModelMeta class. (https://github.com/workfloworchestrator/orchestrator-core/blob/2b2d17368da40027056db20b32a2618014f608eb/orchestrator/domain/base.py#L585C14-L585C14)

The type is `str`, but we also set it to `None` so it's asking for mypy trouble. 1.7.1 complained about it so this code is backwards compatible and 1.7.1 will not complain if we do upgrade to it.

But better would be finding a less hacky solution.

Reply:
@mrijk It's also this `name` attribute that Pydantic 2.x warns about, because it's used as both a class variable and pydantic Field. That problem may or may not be relevant here, just mentioning it.

In practice, name will never be `None` because this only happens for the abstract classes which we don't register. But as Daniel said this is impossible for mypy to derive. There's probably a clever OOP solution for this.